internal/contour/snapshot.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contour
+
+import (
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+	resource "github.com/envoyproxy/go-control-plane/pkg/resource/v2"
+	"github.com/projectcontour/contour/internal/dag"
+	contourxds "github.com/projectcontour/contour/internal/xds"
+	"github.com/sirupsen/logrus"
+)
+
+// SnapshotHandler implements the xDS snapshot cache
+type SnapshotHandler struct {
+	clusterCache   contourxds.Resource
+	listenerCache  contourxds.Resource
+	secretCache    contourxds.Resource
+	endpointsCache contourxds.Resource
+	routeCache     contourxds.Resource
```

Comment:
You could still store all these caches in a map keyed on ResourceType if you wanted, but I'm not exactly sure how they'll eventually be used. Easy to change later though if you do want to do that, so not a big deal.

Reply:
I prefer that approach and not have to map each type in a for loop, but since we need that it's nice to have each cache addressable directly. If that doesn't pan out can refactor later, but I think this will work out nice.